cmd/node_agent/na/nodeagent.go

```diff
@@ -167,8 +162,12 @@ func (na *nodeAgentInternal) Start() error {
 }
 
 func (na *nodeAgentInternal) createRequest() ([]byte, *pb.Request, error) {
+	identity, err := na.pr.GetServiceIdentity(na.config.CertChainFile)
```

Comment:
Is this something we can read once and set it into a field of nodeAgentInternal?

Reply:
good point, done.